File: v1/src/main/java/com/api/v1/car/controllers/CarRegistrationController.java
package com.api.v1.car.controllers;

import com.api.v1.car.dtos.CarRegistrationRequestDto;
import com.api.v1.car.dtos.CarResponseDto;
import com.api.v1.car.services.CarRegistrationService;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("api/v1/cars")
public class CarRegistrationController {

    @Autowired
    private CarRegistrationService service;

    @PostMapping
    @ResponseStatus(value = HttpStatus.CREATED)
    public Mono<CarResponseDto> register(@RequestBody @Valid CarRegistrationRequestDto requestDto) {
        return service.register(requestDto);
    }

}

File: v2/src/main/kotlin/com/api/v2/car/controllers/CarRegistrationController.kt
package com.api.v2.car.controllers

import com.api.v2.car.domain.Car
import com.api.v2.car.services.CarRegistrationService
import jakarta.validation.Valid
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("api/v2/cars")
class CarRegistrationController {

    @Autowired
    private lateinit var service: CarRegistrationService

    @PostMapping
    @ResponseStatus(value = HttpStatus.CREATED)
    suspend fun register(@RequestBody car: @Valid Car) {
        return service.register(car)
    }

}